studiopress/genesis-custom-blocks
# Add editor field UI component

This should be ready for review when https://github.com/studiopress/genesis-custom-blocks/pull/43 is merged, as it's branched off that.

#### Changes
* Uses Rob's HTML and CSS
* Adds the main field grid:
![the-new-field-grid](https://user-images.githubusercontent.com/4063887/99123933-cee81280-25c6-11eb-9819-befc7502ec73.gif)

* The diff is less than it seems. This was branched off https://github.com/studiopress/genesis-custom-blocks/pull/43, but as soon as that's merged, it'll be [much simpler](https://github.com/studiopress/genesis-custom-blocks/compare/add/ui-block-settings...add/edit-block-ui-main).
* For GF-2529

#### Testing instructions
1. Follow steps 1-4 in https://github.com/studiopress/genesis-custom-blocks/pull/43#issue-518208441
1. Expected: it should look like this:
![fields-grid](https://user-images.githubusercontent.com/4063887/99356632-15ac6580-2870-11eb-825f-b6be79906ea9.png)
1. Expected: on clicking the field in the grid, the field panel shows, and you can delete the field:
![on-th-wes](https://user-images.githubusercontent.com/4063887/99356923-a125f680-2870-11eb-9014-81c84526fc9a.gif)
1. Expected: you can add fields and resize them: 
![field-rsa](https://user-images.githubusercontent.com/4063887/99357140-0974d800-2871-11eb-8eaa-987208984940.gif)
1. Expected: you can toggle to 'Inspector fields, but that grid doesn't really work. That's for a later ticket: 
![ijsdf-field](https://user-images.githubusercontent.com/4063887/99357355-61134380-2871-11eb-9616-02645ac1baac.gif)
1. Not in scope: moving the fields in the grid (GF-2528)
